erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -56,9 +56,12 @@ def before_save(self):
 		if not self.on_hold:
 			self.release_date = ''
 
-
 	def invoice_is_blocked(self):
 		return self.on_hold and (not self.release_date or self.release_date > getdate(nowdate()))
+	
+	def on_update(self):
+		self.title = self.supplier
+		frappe.db.set_value("Purchase Invoice",self.name,'title',self.supplier)
```

Comment:
Avoid doing `db.set_value` wherever possible, since that overrides validations and enters the value directly into the database

Once you move this to validate, you won't need the `set_value` call